BlueprintUICommonControls/Sources/ScrollView.swift

```diff
@@ -262,8 +269,53 @@ fileprivate final class ScrollerWrapperView: UIView {
                 self.scrollView.contentOffset.x = -contentInset.left
             }
         }
-
-
     }
-
+    
+    //
+    // MARK: UIView
+    //
+    
+    public override func didMoveToWindow() {
+        super.didMoveToWindow()
+        
+        if self.window != nil {
+            self.setContentInsetWithKeyboardFrame()
+        }
+    }
+    
+    public override func didMoveToSuperview() {
+        super.didMoveToSuperview()
+        
+        if self.superview != nil {
+            self.setContentInsetWithKeyboardFrame()
+        }
+    }
+    
+    //
+    // MARK: Keyboard
+    //
+    
+    private func setContentInsetWithKeyboardFrame() {
+        guard let frame = self.keyboardObserver.currentFrame(in: self) else {
+            return
+        }
+        
+        var inset : CGFloat
+        
+        switch frame {
+        case .notVisible: inset = 0.0
+        case .visible(let frame): inset = (self.bounds.size.height - frame.origin.y)
+        }
+        
+        self.scrollView.contentInset.bottom = inset
+    }
+    
+    //
+    // MARK: KeyboardObserverDelegate
+    //
+    
+    func keyboardFrameWillChange(observer : KeyboardObserver)
+    {
+        self.setContentInsetWithKeyboardFrame()
```

Comment:
Can we have a property that enables this behavior, rather than hardcoding it in?

Reply:
Sure

Reply:
Thoughts on the name? `adjustsContentInsetForKeyboard`? Or could do an enum `KeyboardAdjustmentType` and have a `.none` and `.adjusts`, which reads a bit nicer.

Reply:
Going with `keyboardAdjustmentMode` for now.